internal/cloud/azure.go

```diff
@@ -0,0 +1,209 @@
+/*
+Based on https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=linux#instance-metadata
+*/
+
+package cloud
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	azureApiVersion = "2021-02-01"
+	azureApiAddress = "169.254.169.254"
+	azurePortalUrl  = "https://portal.azure.com/#@SUSERDBillingsuse.onmicrosoft.com/resource"
+)
+
+type AzureMetadata struct {
+	Compute Compute `json:"compute,omitempty" mapstructure:"compute,omitempty"`
+	Network Network `json:"network,omitempty" mapstructure:"network,omitempty"`
+}
+
+type Compute struct {
```

Comment:
This `Compute` struct in the `cloud` package feels to be specific to azure, right? How will we handle this when multiple cloud providers are implemented? should we take these out of `azure.go` maybe into a `cloud.go` file and make them more generic or will each cloud provider implementation include their own structs?

Reply:
Yes, this `compute` is Azure specific, and it is on purpose.
Later on the `cloud.go` integrates this in its `metadata`, which is an `interface`, so it will take anything you give. This time is azure data, but in other providers will be other information.
So it should be fine

Reply:
And yes, each cloud provider will have their own structs depending on how the structure their metadata information